pkg/networkservice/common/mechanisms/kernel/client.go

```diff
@@ -21,19 +21,24 @@ import (
 	"net/url"
 
 	"github.com/golang/protobuf/ptypes/empty"
+	"google.golang.org/grpc"
+
 	"github.com/networkservicemesh/api/pkg/api/networkservice"
 	"github.com/networkservicemesh/api/pkg/api/networkservice/mechanisms/cls"
 	"github.com/networkservicemesh/api/pkg/api/networkservice/mechanisms/kernel"
-	"google.golang.org/grpc"
 
 	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
 )
 
-type kernelMechanismClient struct{}
+type kernelMechanismClient struct {
+	interfaceName string
+}
 
 // NewClient - returns client that sets kernel preferred mechanism
-func NewClient() networkservice.NetworkServiceClient {
-	return &kernelMechanismClient{}
+func NewClient(interfaceName string) networkservice.NetworkServiceClient {
+	return &kernelMechanismClient{
+		interfaceName: interfaceName,
```

Comment:
As I can see we can work with `interfaceName== ""` https://github.com/networkservicemesh/api/blob/e5aee3ba99efe11afe73263b9d2e4396da3da1f7/pkg/api/networkservice/mechanisms/kernel/helpers.go#L92
So I would recommend adding an option for this chain element to set some custom interface name.

Reply:
It is actually an option, user can set `interfaceName== ""` if there is no need for some custom interface name.

Reply:
I think better if the user do not need to pass anything into the constructor.

Reply:
Yep, done